From: Kevin Funk <kfunk@kde.org>
Date: Tue, 22 Dec 2015 14:25:48 +0000
Subject: CMake: KDevplatform integration fixes
X-Git-Tag: v5.5.95
X-Git-Url: http://quickgit.kde.org/?p=plasma-sdk.git&a=commitdiff&h=616cd9c3cabff9b826fdba935b65b8dc45b05ac0
---
CMake: KDevplatform integration fixes

Summary:
* Adapt to kdevplatform changes
* CMake: Simplify kdevplatform lookup

Reviewers: mart

Reviewed By: mart

Differential Revision: https://phabricator.kde.org/D668
---


--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,6 +36,12 @@
     set(BUILD_TESTING OFF CACHE BOOL "Build the testing tree.")
 endif()
 
+find_package(KDevPlatform 4.90.90)
+set_package_properties(KDevPlatform PROPERTIES
+    PURPOSE "Required for plasmate"
+    TYPE OPTIONAL
+)
+
 add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0)
 
 include(KDEInstallDirs)
@@ -56,7 +62,7 @@
 
 # Load CMake, Compiler and InstallDirs settings from KF5 and the following are already somewhat "done" tier1/tier2 libs from kdelibs:
 find_package(KF5 REQUIRED COMPONENTS
-    Archive Completion Config ConfigWidgets CoreAddons DBusAddons Declarative I18n IconThemes
+    Archive Completion Config ConfigWidgets CoreAddons DBusAddons Declarative I18n IconThemes ItemModels
     KIO NewStuff Parts Plasma PlasmaQuick Service TextEditor WidgetsAddons XmlGui WindowSystem)
 
 #########################################################################
@@ -66,18 +72,15 @@
 
 remove_definitions(-DQT_NO_CAST_FROM_ASCII -DQT_STRICT_ITERATORS -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_KEYWORDS)
 
-option(PLASMATE_BUILD_WITH_KDEVPLATFORM "Build the plasmate repository with KDevPlatform Support" OFF)
-message("Plasmate depends on Kdevplatform. So since Kdevplatform has not been released yes, Plasmate's build is being disabled by default for the time being.")
-message("Please pass -DPLASMATE_BUILD_WITH_KDEVPLATFORM=ON in order to build the Plasmate")
-
 add_subdirectory(cuttlefish)
 add_subdirectory(engineexplorer)
 add_subdirectory(plasmoidviewer)
 add_subdirectory(themeexplorer)
 #add_subdirectory(remote-widgets-browser)
 #add_subdirectory(wallpaperviewer)
-if(PLASMATE_BUILD_WITH_KDEVPLATFORM)
+if (KDevPlatform_FOUND)
     add_subdirectory(plasmate)
 endif()
 
+feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
 

--- a/plasmate/CMakeLists.txt
+++ b/plasmate/CMakeLists.txt
@@ -11,16 +11,9 @@
 find_program(GIT_EXECUTABLE_FOUND git)
 add_feature_info(GIT_EXECUTABLE_FOUND "Git" "Distributed Revision Control System" "http://git-scm.com/" FALSE "" "Runtime dependency needed to use the TimeLine feature.")
 
-set( KDEVELOP_VERSION_MAJOR 4 )
-set( KDEVELOP_VERSION_MINOR 90 )
-set( KDEVELOP_VERSION_PATCH 60 )
-# KDevplatform dependency version
-set( KDEVPLATFORM_VERSION "1.${KDEVELOP_VERSION_MINOR}.${KDEVELOP_VERSION_PATCH}" )
-
-find_package(KDevPlatform ${KDEVPLATFORM_VERSION} REQUIRED)
-
 find_package(KF5 REQUIRED COMPONENTS
-    KDELibs4Support)
+    KDELibs4Support
+)
 
 #install( FILES previewer/windowswitcher/thumbnails/konqueror.png
 #previewer/windowswitcher/thumbnails/kmail.png

--- a/plasmate/app/CMakeLists.txt
+++ b/plasmate/app/CMakeLists.txt
@@ -31,8 +31,8 @@
                       KF5::PlasmaQuick
                       KF5::Archive
                       #${KDEPIMLIBS_KPIMUTILS_LIBS}
-                      ${KDEVPLATFORM_INTERFACES_LIBRARIES}
-                      ${KDEVPLATFORM_SHELL_LIBRARIES}
+                      KDev::Interfaces
+                      KDev::Shell
 )
 
 install(TARGETS plasmate ${INSTALL_TARGETS_DEFAULT_ARGS})

--- a/plasmate/plugins/plasmaprojectmanager/CMakeLists.txt
+++ b/plasmate/plugins/plasmaprojectmanager/CMakeLists.txt
@@ -6,9 +6,9 @@
 
 add_library(kdevplasmamanager MODULE ${kdevplasmamanager_SRCS})
 target_link_libraries(kdevplasmamanager
-    KF5::KIOWidgets
     KDev::Project
     KDev::Util
+    KF5::KIOWidgets
     KF5::Plasma
 )
 

--- a/plasmate/plugins/savesystemview/CMakeLists.txt
+++ b/plasmate/plugins/savesystemview/CMakeLists.txt
@@ -6,12 +6,13 @@
 )
 
 add_library(kdevsavesystemview MODULE ${kdevsavesystemview_SRCS})
-target_link_libraries(kdevsavesystemview ${KDEVPLATFORM_INTERFACES_LIBRARIES}
-                                         ${KDEVPLATFORM_PROJECT_LIBRARIES}
-                                         ${KDEVPLATFORM_UTIL_LIBRARIES}
-                                         ${KDEVPLATFORM_VCS_LIBRARIES}
-                                         Qt5::Quick
-                                         KF5::Declarative
+target_link_libraries(kdevsavesystemview
+    KDev::Interfaces
+    KDev::Project
+    KDev::Util
+    KDev::Vcs
+    Qt5::Quick
+    KF5::Declarative
 )
 
 install(TARGETS kdevsavesystemview DESTINATION ${PLUGIN_INSTALL_DIR} )

